### PR TITLE
feat(RELEASE-1179): docs for single component mode

### PR DESCRIPTION
**Pull request by scoheb**
- added in advanced how-tos

### CODE REVIEW

File: docs/modules/ROOT/pages/advanced-how-tos/testing-releasing-single-component.adoc [hunk 1]
```diff
@@ -0,0 +1,200 @@
+= Testing and Releasing a Single Component
+
+By default, Konflux treats applications as a set of dependent components. A example would frontend and backend
```
Comment (arewm) [inline docs/modules/ROOT/pages/advanced-how-tos/testing-releasing-single-component.adoc:3]:
```suggestion
By default, {ProductName} treats applications as a set of dependent components. A example would be frontend and backend
```

File: docs/modules/ROOT/pages/advanced-how-tos/testing-releasing-single-component.adoc [hunk 2]
```diff
@@ -0,0 +1,200 @@
...
+components. Typically, a team would want to build these components, while testing and releasing them together
```
Comment (arewm) [inline docs/modules/ROOT/pages/advanced-how-tos/testing-releasing-single-component.adoc:4]:
```suggestion
components. Typically, a user would want to build these components, while testing and releasing them together
```

File: package-lock.json [hunk 3]
```diff
@@ -1,5 +1,5 @@
 {
-  "name": "konflux-ci-docs",
+  "name": "konflux-upstream-docs",
```
Comment (arewm) [inline package-lock.json:2]: Was this change intentional?

File: docs/modules/ROOT/pages/advanced-how-tos/testing-releasing-single-component.adoc [hunk 4]
```diff
@@ -0,0 +1,200 @@
...
+      pipeline: 4h0m0s
+  policy: e2e-single-component-policy
+----
+
+[NOTE]
+====
+You need to apply the change using *kubectl*
+====
+
+=== Analyzing Test Results and Releases in Single Component Mode
```
Comment (arewm) [inline docs/modules/ROOT/pages/advanced-how-tos/testing-releasing-single-component.adoc:123]: Is this analyzing or just verifying? Since we go over ITS and releases in subcategories, we can probably just call this verifying generically.

File: docs/modules/ROOT/pages/advanced-how-tos/testing-releasing-single-component.adoc [hunk 5]
```diff
@@ -0,0 +1,200 @@
...
+
+* Once your changes have been applied, you can verify that the mode is in fact activated.
+
+==== Verifying Enterprise Contract Integration Test Scenario results
```
Comment (arewm) [inline docs/modules/ROOT/pages/advanced-how-tos/testing-releasing-single-component.adoc:127]: Is this only for the EC ITS or ITS generally?

File: docs/modules/ROOT/pages/advanced-how-tos/testing-releasing-single-component.adoc [hunk 6]
```diff
@@ -0,0 +1,200 @@
...
+
+* Navigate to the logs of your Integration Test Scenario pipeline.
+* Locate the *reduce* step in the log.
+* It should resemble the following output:
+
+[source]
+--
+[reduce] Single Component mode? true
```
Comment (arewm) [inline docs/modules/ROOT/pages/advanced-how-tos/testing-releasing-single-component.adoc:135]: For this and the next snippet, is the content relevant or just the fact that the reduce parts ran? It might be simpler to parse if we remove the unnecessary details specific to your run.

File: docs/modules/ROOT/pages/advanced-how-tos/testing-releasing-single-component.adoc [hunk 7]
```diff
@@ -0,0 +1,200 @@
...
+kind: IntegrationTestScenario
+metadata:
+  name: grouping-enterprise-contract-single-component
+  namespace: shebert-tenant
+spec:
+  application: grouping
+  contexts:
+    - description: Application testing
+      name: application
+  params:
```
Comment (arewm) [inline docs/modules/ROOT/pages/advanced-how-tos/testing-releasing-single-component.adoc:47]: Is this the only part that is new? If so, we can just use a snippet like we do for defining build customizations so that they don’t need to figure out what information they don’t need to include.

File: docs/modules/ROOT/pages/advanced-how-tos/testing-releasing-single-component.adoc [hunk 8]
```diff
@@ -0,0 +1,200 @@
...
+
+* Alternatively, you may edit the Integration Test Scenario using the HAC UI as follows:
+
+image::single-component-integration-test-scenario.png[role="border" alt="Editing your Integration Test Scenario"]
+
+=== Enabling Single Component Mode for Releases
+
+Single Component Mode can be enabled for Releases by making a change to the ReleasePlanAdmission custom resource.
+
+* Update the custom resource by adding the field *singleComponentMode* under _.spec.data_ with a value of *true*
```
Comment (arewm) [inline docs/modules/ROOT/pages/advanced-how-tos/testing-releasing-single-component.adoc:74]: You can also add the annotation to the RPA CR page with a description and link the two pages instead.

**Review by arewm — CHANGES_REQUESTED** [review]

File: docs/modules/ROOT/pages/advanced-how-tos/testing-releasing-single-component.adoc (hunk 7 above)
Comment (scoheb) [reply docs/modules/ROOT/pages/advanced-how-tos/testing-releasing-single-component.adoc:38]: fixed snippet

File: docs/modules/ROOT/pages/advanced-how-tos/testing-releasing-single-component.adoc (hunk 6 above)
Comment (scoheb) [reply docs/modules/ROOT/pages/advanced-how-tos/testing-releasing-single-component.adoc:83]: removed some of the snippet

File: docs/modules/ROOT/pages/advanced-how-tos/testing-releasing-single-component.adoc (hunk 5 above)
Comment (scoheb) [reply docs/modules/ROOT/pages/advanced-how-tos/testing-releasing-single-component.adoc:75]: EC ITS. I updated the next line to clarify

File: docs/modules/ROOT/pages/advanced-how-tos/testing-releasing-single-component.adoc (hunk 4 above)
Comment (scoheb) [reply docs/modules/ROOT/pages/advanced-how-tos/testing-releasing-single-component.adoc]: fixed

File: package-lock.json (hunk 3 above)
Comment (scoheb) [reply package-lock.json]: guess it was due to my repo name...fixed

File: docs/modules/ROOT/pages/advanced-how-tos/testing-releasing-single-component.adoc (hunk 2 above)
Comment (scoheb) [reply docs/modules/ROOT/pages/advanced-how-tos/testing-releasing-single-component.adoc]: fixed

File: docs/modules/ROOT/pages/advanced-how-tos/testing-releasing-single-component.adoc (hunk 1 above)
Comment (scoheb) [reply docs/modules/ROOT/pages/advanced-how-tos/testing-releasing-single-component.adoc]: fixed

File: docs/modules/ROOT/pages/advanced-how-tos/testing-releasing-single-component.adoc [hunk 9]
```diff
@@ -0,0 +1,110 @@
...
+
+* You have an application that contains 5 components.
+* You need to release a single component to fix a bug that a customer is experiencing.
+* While attempting to release, another component has a new violation therefore causing the Enterprise Contract
+policy validation to fail.
+** This violation may be due to a rule that used to be a warning but has expired and is now a violation.
+
+Wouldn't it be nice to be able to ignore that failing component and simply release the component you are targeting?
+
+Enter **Single Component Mode**!
```
[the remark this answers is not in the document]
Comment (scoheb) [reply docs/modules/ROOT/pages/advanced-how-tos/testing-releasing-single-component.adoc]: I am easy. fixed

File: docs/modules/ROOT/pages/advanced-how-tos/testing-releasing-single-component.adoc [hunk 10]
```diff
@@ -0,0 +1,101 @@
...
+== Enabling Single Component Mode
+
+Single component mode works by analyzing the Snapshot that is being used for Integration Tests and Releasing.
+A Snapshot created due to a single component being built is the only Snapshot type that may be used for this
+mode.
+
+=== Enabling Single Component Mode for Enterprise Contract Integration Test Scenarios
+
+Single Component Mode can be enabled either by making a change to the Integration Test Scenario custom resource
+or by using the HAC UI.
```
Comment (arewm) [inline docs/modules/ROOT/pages/advanced-how-tos/testing-releasing-single-component.adoc:21]:
```suggestion
or by using the UI.
```

**Review by arewm — COMMENTED** [review]
Looks good. I meant to include this small change in the prior review.

File: docs/modules/ROOT/pages/advanced-how-tos/testing-releasing-single-component.adoc (hunk 10 above)
Comment (scoheb) [reply docs/modules/ROOT/pages/advanced-how-tos/testing-releasing-single-component.adoc]: fixed

**Review by arewm — APPROVED** [review]
